test/hotspot/jtreg/runtime/cds/appcds/OldSuperInf.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+/*
+ * @test
+ * @summary CDS support of old classes with major version < JDK_6 (50) for static archive.
+ *          Test with old super interface.
+ * @requires vm.cds
+ * @library /test/lib
+ * @compile test-classes/OldInf.jasm
+ * @compile test-classes/ChildOldInf.java
+ * @compile test-classes/GChild2.java
+ * @compile test-classes/OldSuperInfApp.java
+ * @run driver OldSuperInf
+ */
+
+import jdk.test.lib.cds.CDSTestUtils;
+import jdk.test.lib.process.OutputAnalyzer;
+
+public class OldSuperInf {
+
+    public static void main(String[] args) throws Exception {
+        String mainClass = "OldSuperInfApp";
+        String namePrefix = "oldsuperinf";
+        String appClasses[] = TestCommon.list("OldInf", "ChildOldInf", "GChild2", mainClass);
+        JarBuilder.build(namePrefix, appClasses);
+
+        String appJar = TestCommon.getTestJar(namePrefix + ".jar");
+        String archiveName = namePrefix + ".jsa";
+
+        boolean dynamicMode = CDSTestUtils.DYNAMIC_DUMP;
+
+        // create archive with class list
+        OutputAnalyzer output = TestCommon.dump(appJar, appClasses, "-Xlog:class+load,cds=debug,verification=trace");
+        TestCommon.checkExecReturn(output, 0,
+                                   dynamicMode ? true : false,
```

Comment:
Why use a ternary operator, is it the variable itself OK here?

Reply:
I prefer to leave it as is because the third arg to `TestCommon.checkExecReturn()` is about whether or not to look for the strings in the subsequent args.

Thanks for your review.